File: DynamoDB-SDK-Examples/python/WorkingWithItems/batch_write.py
from __future__ import print_function # Python 2/3 compatibility
import boto3, json, decimal
from boto3.dynamodb.conditions import Key, Attr
from botocore.exceptions import ClientError

dynamodb = boto3.resource('dynamodb', region_name='us-west-2') #replace region name with your target region

table = dynamodb.Table("RetailDatabase") #replace table name with your own

# Batch write 3 items into the table
with table.batch_writer() as batch:
    batch.put_item(
        Item={
            'pk': 'vikram.johnson@somewhere.com',
            'sk': 'metadata',
            'username': 'vikj',
            'first_name': 'Vikram',
            'last_name': 'Johnson',
            'name': 'Vikram Johnson',
            'age': 31,
            'address': {
                'road': '89105 Bakken Rd',
                'city': 'Greenbank',
                'pcode': 98253,
                'state': 'WA',
                'country': 'USA'
            }
        }
    )
    batch.put_item(
        Item={
            'pk': 'jose.schneller@somewhere.com',
            'sk': 'metadata',
            'username': 'joses',
            'first_name': 'Jose',
            'last_name': 'Schneller',
            'name': 'Jose Schneller',
            'age': 27,
            'address': {
                'road': '12341 Fish Rd',
                'city': 'Freeland',
                'pcode': 98249,
                'state': 'WA',
                'country': 'USA'
            }
        }
    )
    batch.put_item(
        Item={
            'pk': 'helga.ramirez@somewhere.com',
            'sk': 'metadata',
            'username': 'helgar',
            'first_name': 'Helga',
            'last_name': 'Ramirez',
            'name': 'Helga Ramirez',
            'age': 48,
            'address': {
                'road': '45678 Deer Lake Rd',
                'city': 'Clinton',
                'pcode': 98236,
                'state': 'WA',
                'country': 'USA'
            }
        }
    )

File: DynamoDB-SDK-Examples/node.js/WorkingWithItems/batch-write.js
const AWS = require("aws-sdk");

AWS.config.update({ region: "us-west-2" });

const documentClient = new AWS.DynamoDB.DocumentClient();

// Batch write 2 items into the table
async function batchWriteItem() {
  const params = {
    RequestItems: {
      RetailDatabase: [
        {
          PutRequest: {
            Item: {
              pk: "vikram.johnson@somewhere.com",
              sk: "metadata",
              username: "vikj",
              first_name: "Vikram",
              last_name: "Johnson",
              name: "Vikram Johnson",
              age: 31,
              address: {
                road: "89105 Bakken Rd",
                city: "Greenbank",
                pcode: 98253,
                state: "WA",
                country: "USA",
              },
            },
          },
        },
        {
          PutRequest: {
            Item: {
              pk: "jose.schneller@somewhere.com",
              sk: "metadata",
              username: "joses",
              first_name: "Jose",
              last_name: "Schneller",
              name: "Jose Schneller",
              age: 27,
              address: {
                road: "12341 Fish Rd",
                city: "Freeland",
                pcode: 98249,
                state: "WA",
                country: "USA",
              },
            },
          },
        },
      ],
    },
  };

  const response = await documentClient.batchWrite(params).promise();
  return response;
}

batchWriteItem()
  .then(() => console.log("BatchWriteItem succeeded"))
  .catch((error) => console.error(JSON.stringify(error, null, 2)));
